src/client.js

```diff
@@ -60,6 +60,16 @@ var _defaultOptions = {
 
   // The prefix of any api calls. e.g. https://taskcluster.net/api/
   rootUrl: process.env.TASKCLUSTER_ROOT,
+
+  // Fake methods, if truthy this will produce a fake client object.
+  // Methods called won't make expected HTTP requests, but instead:
+  //   1. Add arguments to `Client.fakeCalls.<method>.push({...params, payload, query})`
+  //   2. Invoke and return `fake.<method>(...args)`, if `fake.<method>`
+  //      is defined, otherwise null is returned.
```

Comment:
I'd prefer this throw an exception..

Reply:
Why?

Isn't recording the calls made a solid default fake implementation?

Or will we only rarely be able to use such a fake implementation for anything? I suspect we would be able to make a lot of assertions on `MyClient.fakeCalls.<method>` and get pretty far in most cases, without having to write custom fake implementations.

Reply:
Because it's quite possible to forget to fake a method and having it silently return null would leave me thinking it's not called, or might lead to weird errors when a response is expected.  Using a non-faked method on a faked client is definitely a misconfiguration of the tests, so it should be announced loudly, not quietly turned into more subtle bugs.